clients/mocks.go

```diff
@@ -8,33 +8,50 @@ type (
 	GatekeeperMock struct {
 		expectedPermissions Permissions
 		expectedError       error
+		UserIDs             []string
 	}
 	SeagullMock struct{}
 )
 
 //A mock of the Gatekeeper interface
 func NewGatekeeperMock(expectedPermissions Permissions, expectedError error) *GatekeeperMock {
-	return &GatekeeperMock{expectedPermissions, expectedError}
+	return &GatekeeperMock{expectedPermissions, expectedError, []string{}}
+}
+func (mock *GatekeeperMock) SetExpected(expectedPermissions Permissions, expectedError error) {
+	mock.expectedPermissions = expectedPermissions
+	mock.expectedError = expectedError
 }
 
 func (mock *GatekeeperMock) UserInGroup(userID, groupID string) (Permissions, error) {
 	if mock.expectedPermissions != nil || mock.expectedError != nil {
 		return mock.expectedPermissions, mock.expectedError
-	} else {
-		return Permissions{userID: Allowed}, nil
 	}
+	return Permissions{"root": Allowed}, nil
 }
 
 func (mock *GatekeeperMock) UsersInGroup(groupID string) (UsersPermissions, error) {
 	if mock.expectedPermissions != nil || mock.expectedError != nil {
 		return UsersPermissions{groupID: mock.expectedPermissions}, mock.expectedError
-	} else {
-		return UsersPermissions{groupID: Permissions{groupID: Allowed}}, nil
 	}
+	return UsersPermissions{groupID: Permissions{groupID: Allowed}}, nil
+}
+
+func (mock *GatekeeperMock) GroupsForUser(userID string) (UsersPermissions, error) {
```

Comment:
like discussed, I think the structure of the returned value is not exactly what Gatekeeper would send.
Gatekeeper would send a "root" permission for the _userID_ + 1 permission for each user that shared data with this _userID_. By default gatekeeper is sending a "note"+"view" permissions for each users. If x and y shared with z:

{
    "z": {
        "root": {}
    },
    "x": {
        "note": {},
        "view": {}
    },
    "y": {
        "note": {},
        "view": {}
    }
}

If no one shared to z:

{
    "z": {
        "root": {}
    }
}

Reply:
Last commit should fix it (when UserIds is set)

Reply:
yep, I can see that :)